### ISSUE TITLE
bison compilation fails on CentOS 7.9.2009

### ISSUE THREAD

**Issue by dgautam** [jmillikin/rules_bison#8]
I tried to use rules_bison/rules_flex on centos-7.9. But this is continuously failing for following errors. same rules compile well for ubuntu-20.04. I tried setting up BAZEL_COPTS="-std=c99", but still same output. I am using bazel-4.0.0

```
/home/dgautam/.cache/bazel/_bazel_dgautam/install/1a4a2fac02d50c77031d44c0d91b8920/process-wrapper '--timeout=0' '--kill_delay=15' /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections -fdata-sections -MD -MF bazel-out/host/bin/external/bison_v3.3.2/_objs/bison_lib/AnnotationList.d '-frandom-seed=bazel-out/host/bin/external/bison_v3.3.2/_objs/bison_lib/AnnotationList.o' -iquote external/bison_v3.3.2 -iquote bazel-out/host/bin/external/bison_v3.3.2 -Ibazel-out/host/bin/external/bison_v3.3.2/_virtual_includes/bison_lib -isystem external/bison_v3.3.2 -isystem bazel-out/host/bin/external/bison_v3.3.2 -isystem external/bison_v3.3.2/bison-lib -isystem bazel-out/host/bin/external/bison_v3.3.2/bison-lib -isystem external/bison_v3.3.2/gnulib/lib -isystem bazel-out/host/bin/external/bison_v3.3.2/gnulib/lib -isystem external/bison_v3.3.2/gnulib/config-linux -isystem bazel-out/host/bin/external/bison_v3.3.2/gnulib/config-linux -g0 -fno-canonical-system-headers -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c external/bison_v3.3.2/src/AnnotationList.c -o bazel-out/host/bin/external/bison_v3.3.2/_objs/bison_lib/AnnotationList.o)
external/bison_v3.3.2/src/AnnotationList.c: In function 'AnnotationList__compute_conflicted_tokens':
external/bison_v3.3.2/src/AnnotationList.c:173:3: error: 'for' loop initial declarations are only allowed in C99 mode
   for (int i = 0; i < reds->num; ++i)
   ^
external/bison_v3.3.2/src/AnnotationList.c:173:3: note: use option -std=c99 or -std=gnu99 to compile your code
external/bison_v3.3.2/src/AnnotationList.c: In function 'AnnotationList__computePredecessorAnnotations':
external/bison_v3.3.2/src/AnnotationList.c:227:3: error: 'for' loop initial declarations are only allowed in C99 mode
   for (state **predecessor = predecessors[s->number]; *predecessor; ++predecessor)
   ^
external/bison_v3.3.2/src/AnnotationList.c:236:9: error: 'for' loop initial declarations are only allowed in C99 mode
         for (ContributionIndex ci = 0; ci < self->inadequacyNode->contributionCount; ++ci)
         ^
external/bison_v3.3.2/src/AnnotationList.c:334:25: error: 'for' loop initial declarations are only allowed in C99 mode
                         for (size_t j = 0; j < (*predecessor)->nitems; ++j)
                         ^
external/bison_v3.3.2/src/AnnotationList.c:374:13: error: 'for' loop initial declarations are only allowed in C99 mode
             for (size_t i = 0; i < (*predecessor)->nitems; ++i)
             ^
external/bison_v3.3.2/src/AnnotationList.c: In function 'AnnotationList__compute_from_inadequacies':
external/bison_v3.3.2/src/AnnotationList.c:436:9: error: 'for' loop initial declarations are only allowed in C99 mode
         for (int rule_i = 0; rule_i < s->reductions->num; ++rule_i)
         ^
external/bison_v3.3.2/src/AnnotationList.c:452:9: error: 'for' loop initial declarations are only allowed in C99 mode
         for (int rule_i = 0; rule_i < s->reductions->num; ++rule_i)
         ^
external/bison_v3.3.2/src/AnnotationList.c: In function 'AnnotationList__debug':
external/bison_v3.3.2/src/AnnotationList.c:574:7: error: 'for' loop initial declarations are only allowed in C99 mode
       for (int j = 0; j < spaces; ++j)
       ^
external/bison_v3.3.2/src/AnnotationList.c:581:9: error: 'for' loop initial declarations are only allowed in C99 mode
         for (ContributionIndex ci = 0; ci < a->inadequacyNode->contributionCount; ++ci)
         ^
external/bison_v3.3.2/src/AnnotationList.c:586:13: error: 'for' loop initial declarations are only allowed in C99 mode
             for (int j = 0; j < spaces+2; ++j)
             ^
external/bison_v3.3.2/src/AnnotationList.c: In function 'AnnotationList__computeLookaheadFilter':
external/bison_v3.3.2/src/AnnotationList.c:622:5: error: 'for' loop initial declarations are only allowed in C99 mode
     for (ContributionIndex ci = 0; ci < self->inadequacyNode->contributionCount; ++ci)
     ^
external/bison_v3.3.2/src/AnnotationList.c: In function 'AnnotationList__computeDominantContribution':
external/bison_v3.3.2/src/AnnotationList.c:777:3: error: 'for' loop initial declarations are only allowed in C99 mode
   for (ContributionIndex ci = 0; ci < self->inadequacyNode->contributionCount; ++ci)
   ^
INFO: Elapsed time: 1.718s, Critical Path: 0.25s
INFO: 13 processes: 10 internal, 3 processwrapper-sandbox.
FAILED: Build did NOT complete successfully
```

**Comment by dgautam** [comment]
I used following rule:
```
genrule(
    name = "sandeshl_flex",
    srcs = ["sandeshl.ll"],
    outs = ["sandeshl.cc"],
    cmd = "M4=$(M4) $(FLEX) -w -t $< > $@",
    toolchains = [
        "@rules_flex//flex:current_flex_toolchain",
        "@rules_m4//m4:current_m4_toolchain",
    ],
)
```

**Comment by jmillikin** [comment]
Fixed in deb3a83a6c88beec88c1e31450f0ab2ec415a419

It turns out that CentOS 7 runs an extremely old version of GCC (v4.8, from 2013), which doesn't enable C99 features by default. The above commit adds `-std=c99` on non-Windows builds as a workaround.

For future reference, it's also possible to add custom C/C++ compiler options with the `extra_cflags=` attribute on `rules_bison` repository rules.
